app/models/custom_value.rb

```diff
@@ -38,6 +38,32 @@ class CustomValue < ActiveRecord::Base
 
   after_initialize :set_default_value
 
+  # returns the value of this custom value, but converts it according to the field_format
+  # of the custom field beforehand
+  def typed_value
+    return nil if value.nil?
+
+    # strings may be blank, all other types can only be nil if their value is blank
+    return nil if value.blank? && !(['string', 'text'].include?(custom_field.field_format))
+
+    case custom_field.field_format
+    when 'int'
+      value.to_i
+    when 'float'
+      value.to_f
+    when 'date'
+      Date.iso8601(value)
+    when 'bool'
+      value == '1'
+    when 'user'
+      User.find(value)
+    when 'version'
+      Version.find(value)
```

Comment:
This will cause 500 errors in cases where a version/user is deleted after a value has been set. For users, we should double check whether user deletion does rewrite such values to the DeletedUser. For versions, no such mechanism exists. We might be forced to add it now.